File: cdk-from-cfn-testing/expected/condition_sub/golang/ConditionSubStack.go
package main

import (
	"fmt"

	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	sns "github.com/aws/aws-cdk-go/awscdk/v2/awssns"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type ConditionSubStackProps struct {
	cdk.StackProps
	EnvName *string
}

/// Exercises Fn::Sub inside a Condition
type ConditionSubStack struct {
	cdk.Stack
}

func NewConditionSubStack(scope constructs.Construct, id string, props *ConditionSubStackProps) *ConditionSubStack {
	var sprops cdk.StackProps
	if props != nil {
		sprops = props.StackProps
	}
	stack := cdk.NewStack(scope, &id, &sprops)

	hasCustomSuffix := jsii.String(fmt.Sprintf("%v-suffix", props.EnvName)) == jsii.String("prod-suffix")

	sns.NewCfnTopic(
		stack,
		jsii.String("Topic"),
		&sns.CfnTopicProps{
		},
	)

	return &ConditionSubStack{
		Stack: stack,
	}
}

File: cdk-from-cfn-testing/expected/condition_sub/java/src/main/java/com/myorg/ConditionSubStack.java
package com.myorg;

import software.constructs.Construct;

import java.util.*;
import software.amazon.awscdk.CfnMapping;
import software.amazon.awscdk.CfnTag;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;

import software.amazon.awscdk.*;
import software.amazon.awscdk.services.sns.*;

class ConditionSubStack extends Stack {
    public ConditionSubStack(final Construct scope, final String id) {
        super(scope, id, null);
    }

    public ConditionSubStack(final Construct scope, final String id, final StackProps props) {
        this(scope, id, props, null);
    }

    public ConditionSubStack(final Construct scope, final String id, final StackProps props,
            String envName) {
        super(scope, id, props);


        Boolean hasCustomSuffix = (envName + "-suffix").equals("prod-suffix");

        Optional<CfnTopic> topic = hasCustomSuffix ? Optional.of(CfnTopic.Builder.create(this, "Topic")
                .build()) : Optional.empty();

    }
}
